### PR TITLE
Improve indentation of next line when using CTRL-o in terminal.

### LINKED ISSUE #9588: In the PTK terminal, control-o should indent the newly created line
Suppose I have:

```
In [5]: def f():
   ...:     [CURSOR]a = 1           
```

If I press control-O to open up a new line, I get:

```
In [5]: def f():
   ...:     [CURSOR]
   ...: a = 1
```

I would prefer to get:

```
In [5]: def f():
   ...:     [CURSOR]
   ...:     a = 1
```

### PR DESCRIPTION
**Pull request by Carreau**
Closes #9588

Similar to #10264, but with a simpler logic as we never execute.

--- 

I think that @takluyver might want to have a look as well as @gforsyth.

We can probably get rid of the need for an InputSplitter instance as we process all at one. 
Would it make sens to in general remove the stateful part of InutSplitter, and open a separate issue for this ?

### CODE REVIEW

**Comment by takluyver** [comment]
I merged #10264 before seeing this. Is it still relevant?

**Comment by Carreau** [comment]
Yep. #10264 is (was) the behavior of Enter (ie, execute or insert new line and indent moving the cursor.
This is insert newline and indent _after_ the cursor.

File: IPython/terminal/shortcuts.py [hunk 1]
```diff
@@ -178,6 +198,38 @@ def newline_with_copy_margin(event):
...
+
+        text = d.text[:d.cursor_position] + '\n'
+
+        _, indent = inputsplitter.check_complete(text)
+        b.insert_text('\n' + (' ' * (indent or 0)))
+        b.cursor_up(count=1)
+        cursor_end_pos = b.document.cursor_position_col
+        if cursor_start_pos != cursor_end_pos:
+            pos_diff = cursor_start_pos - cursor_end_pos
+            b.cursor_right(count=pos_diff)
```
Comment (takluyver) [inline IPython/terminal/shortcuts.py]: Looking at the prompt_toolkit source code, `.insert_text()` can take a `move_cursor=False` parameter, which would simplify this a fair bit.

Comment (takluyver) [inline IPython/terminal/shortcuts.py]: And that has been there [since 1.0.4], which is the minimum we require, so we should be fine to use it.

Comment (takluyver) [inline IPython/terminal/shortcuts.py]: That link was: https://github.com/jonathanslenders/python-prompt-toolkit/blob/1.0.4/prompt_toolkit/buffer.py#L913

**Comment by takluyver** [comment]
I suspect there are bits of InputSplitter's API we can simplify now, but that would have to be done very carefully. If you do create an issue, please assign it to me.

File: IPython/terminal/shortcuts.py (hunk 1 above)
Comment (Carreau) [inline IPython/terminal/shortcuts.py]: Well spotted. Updated accordingly.

**Pull request merged**